runtime/protocol/http-client-engines/http-client-engine-okhttp/api/http-client-engine-okhttp.api

```diff
@@ -14,7 +65,7 @@ public final class aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngine$Com
 
 public final class aws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngineConfig : aws/smithy/kotlin/runtime/http/engine/HttpClientEngineConfigImpl {
 	public static final field Companion Laws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngineConfig$Companion;
-	public synthetic fun <init> (Laws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngineConfig$Builder;Lkotlin/jvm/internal/DefaultConstructorMarker;)V
+	public fun <init> (Laws/smithy/kotlin/runtime/http/engine/okhttp/OkHttpEngineConfig$Builder;)V
```

Comment:
Moving the constructor from `internal` to `@InternalApi public` results in this "API break", I think it's safe in practice

Reply:
This is definitely a breaking change. If an SDK client built against the latest engine attempts to call the constructor and the _old_ engine is actually on the classpath, users will see a `NoSuchMethodError`. Method signatures used by callers are resolved when the caller is built. This change is _source-safe_ but not _binary-safe_.

Can we leave the previous `internal` constructor and add a new factory method instead?

Reply:
Thanks for catching that and reminding me. Actually it seems like this change was not necessary after all. `OkHttpEngine4` only uses the public constructor here.